tedana/utils.py

```diff
@@ -49,88 +49,159 @@ def reshape_niimg(data):
+            keep all of them.
+    c.  For the exemplar voxel from the first echo, calculate 1/3 of the mean value for each echo.
+
+        -   This is the threshold for "good" data.
+        -   The 1/3 value is arbitrary.
+        -   If there was more than one exemplar voxel, retain the the highest value for each echo.
+    d.  For each voxel, count the number of echoes that have a mean value greater than the
+        corresponding echo's threshold.
+
+    Decay
+
+    Determine the echo at which the signal stops decreasing for each voxel.
+    If a voxel's signal stops decreasing as echo time increases, then we can infer that the
+    voxel has either fully dephased (i.e., "bottomed out") or been contaminated by noise.
+    This essentially identifies the last echo with "good" data.
+    For a scan that collects many echoes for T2* estimation or has a relatively short echo
+    spacing, it is possible that a later echo will have a higher value,
+    but the overall trend still shows a decay.
+    This method should not be used in those situations.
+
+    The element-wise minimum value between any selected methods is used to construct the adaptive
+    mask.
     """
     RepLGR.info(
-        "An adaptive mask was then generated, in which each voxel's "
-        "value reflects the number of echoes with 'good' data."
+        f"An adaptive mask was then generated using the {'+'.join(methods)} method(s), "
+        "in which each voxel's value reflects the number of echoes with 'good' data."
     )
-
-    # take temporal mean of echos and extract non-zero values in first echo
-    echo_means = data.mean(axis=-1)  # temporal mean of echos
-    first_echo = echo_means[echo_means[:, 0] != 0, 0]
-
-    # get 33rd %ile of `first_echo` and find corresponding index
-    # NOTE: percentile is arbitrary
-    # TODO: "interpolation" param changed to "method" in numpy 1.22.0
-    #       confirm method="higher" is the same as interpolation="higher"
-    #       Current minimum version for numpy in tedana is 1.16 where
-    #       there is no "method" parameter. Either wait until we bump
-    #       our minimum numpy version to 1.22 or add a version check
-    #       or try/catch statement.
-    perc = np.percentile(first_echo, 33, interpolation="higher")
-    perc_val = echo_means[:, 0] == perc
-
-    # extract values from all echos at relevant index
-    # NOTE: threshold of 1/3 voxel value is arbitrary
-    lthrs = np.squeeze(echo_means[perc_val].T) / 3
-
-    # if multiple samples were extracted per echo, keep the one w/the highest signal
-    if lthrs.ndim > 1:
-        lthrs = lthrs[:, lthrs.sum(axis=0).argmax()]
-
-    # determine samples where absolute value is greater than echo-specific thresholds
-    # and count # of echos that pass criterion
-    masksum = (np.abs(echo_means) > lthrs).sum(axis=-1)
+    assert methods, "No methods provided for adaptive mask generation."
```

Comment:
```suggestion
    if (methods is None) or (len(methods) == 1 and methods[0].lower() == "none"):
        LGR.warning(
            "No methods provided for adaptive mask generation. "
            "Only removing voxels with negative or NaN values"
        )
        RepLGR.info(
            "An adaptive mask was then generated that retained echoes with negative or NaN values."
        )
    else:
        RepLGR.info(
            f"An adaptive mask was then generated using the {'+'.join(methods)} method(s), "
            "in which each voxel's value reflects the number of echoes with 'good' data."
        )
```
This wasn't letting me include the RepLGR line above, but it should be part of this suggestion as well. This is going to make sure the report text is parsed correctly if None is an input. If `methods` is None or if `none` is the only element in the `methods` list, there is one bit of output text and there's another text if `decay` and/or `dropout` are included. If someone decides to input `["decay", "dropout", "none"]` it will just ignore the `none`.